test/chaat/fixture.clj

```diff
@@ -0,0 +1,38 @@
+(ns chaat.fixture
+  (:require [chaat.config :as config]
+            [com.stuartsierra.component :as component]
+            [next.jdbc.connection :as conn]
+            [chaat.app :as app]
+            [ragtime.jdbc]
+            [ragtime.repl]
+            [chaat.migrations :as migrations])
+  (:import (com.zaxxer.hikari HikariDataSource)))
+
+(defn new-test-system [config]
+  (component/system-map
+   :server (component/using (app/new-http-server config) {:db :db})
+   :db (conn/component HikariDataSource (:pg-test-dbspec config))))
+
+(def test-system-config {:options {:port (config/get-local-test-port)
+                                   :join? false}
+                         :pg-test-dbspec (config/get-pg-test-dbspec)})
+
+(def test-system (new-test-system test-system-config))
+
+(defn start-test-system []
+  (alter-var-root #'test-system component/start))
+
+(defn stop-test-system []
+  (alter-var-root #'test-system component/stop))
+
+;; using a full system for all tests now. this needs to be optimized.
```

Comment:
Full system: ring server + database. 
In many of these tests, I don't need the local server to be running, hence I don't need a full system. Unless I'm doing something like a end -> end test.
By optimize I mean I will only setup and run the parts of the system essential to the test, instead of the entire system. I will do this in a future PR.
Regarding your final question, I'm not sure if there are issues with running the full system, but it is unnecessary.